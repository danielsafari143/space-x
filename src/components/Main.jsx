import { Outlet } from 'react-router';
import Navbar from './Navbar';

const Main = () => (
  <>
    <Navbar />
    <Outlet />
  </>
);

export default Main;
