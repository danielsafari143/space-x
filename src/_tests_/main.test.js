import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Main from '../components/Main';

describe('Main component', () => {
  test('Renders correctly', () => {
    const tree = render(
      <MemoryRouter>
        <Main />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
