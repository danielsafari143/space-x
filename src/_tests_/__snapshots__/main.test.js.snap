// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="header"
        >
          <h1
            class="title"
          >
            <img
              alt="logo"
              src="planet.png"
            />
            Space Travelers
          </h1>
          <nav>
            <a
              aria-current="page"
              class="active link"
              href="/"
            >
              Rockets
            </a>
            <a
              class="link"
              href="/missions"
            >
              Missions
            </a>
            <hr
              class="nav-hr"
            />
            <a
              class="link"
              href="/myprofile"
            >
              My Profile
            </a>
          </nav>
        </div>
        <hr
          class="header-hr"
        />
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="header"
      >
        <h1
          class="title"
        >
          <img
            alt="logo"
            src="planet.png"
          />
          Space Travelers
        </h1>
        <nav>
          <a
            aria-current="page"
            class="active link"
            href="/"
          >
            Rockets
          </a>
          <a
            class="link"
            href="/missions"
          >
            Missions
          </a>
          <hr
            class="nav-hr"
          />
          <a
            class="link"
            href="/myprofile"
          >
            My Profile
          </a>
        </nav>
      </div>
      <hr
        class="header-hr"
      />
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
